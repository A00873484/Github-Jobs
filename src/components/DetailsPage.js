import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import parse from 'html-react-parser'

function DetailsPage ({ job, history }) {
  const backToHome = () => {
    history.push('/')
  }
  if (job.id !== 0) {
    return (
      <div className='col'>
        <div className='page-header py-3'>
          <div onClick={backToHome}>
            <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" className="bi bi-house-fill ml-3" viewBox="0 0 16 16">
              <path fillRule="evenodd" d="M8 3.293l6 6V13.5a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 13.5V9.293l6-6zm5-.793V6l-2-2V2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5z"/>
              <path fillRule="evenodd" d="M7.293 1.5a1 1 0 0 1 1.414 0l6.647 6.646a.5.5 0 0 1-.708.708L8 2.207 1.354 8.854a.5.5 0 1 1-.708-.708L7.293 1.5z"/>
            </svg>
          </div>
        </div>
          <div className='card mb-4 px-4 py-4'>
              <div className='mb-4 d-flex justify-content-center'>
                  <img src={job.company_logo} alt={job.company} className="w-25 img-fluid"/>
              </div>
              <h3 className="mb-3">{job.company}</h3>
              <div className="pb-4">
                  <a className='text-secondary' href={job.company_url}>
                      {job.company_url}
                  </a>
                  <div id='location'>{job.location}</div>
                  <div id='title'>{job.title}</div>
                  <div id='type'>{job.type}</div>
              </div>
              <div className='blog-comments__content'>
                  {parse(job.description)}
              </div>
              <span id='date' className='text-mutes'>{new Date(job.created_at).toDateString()}</span>
          </div>
      </div>
    )
  }
  return null
}

DetailsPage.propTypes = {
  job: PropTypes.object,
  history: PropTypes.object
}

function mapStateToProps ({ jobs }, props) {
  const { id } = props.match.params
  if (!jobs[id]) {
    props.history.push('/')
    return { job: { id: 0 } }
  } else {
    return {
      job: jobs[id],
      history: props.history
    }
  }
}

export default connect(mapStateToProps)(DetailsPage)
