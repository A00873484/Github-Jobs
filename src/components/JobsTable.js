import { Col, Card, CardHeader, CardBody } from 'shards-react'
import { withRouter } from 'react-router-dom'
import React from 'react'
import PropTypes from 'prop-types'

function Table ({ jobs, history }) {
  const jobClicked = (id) => {
    history.push(`/details/${id}`)
  }

  return (
    <Col>
        <Card small className='mb-4'>
            <CardHeader className='border-bottom'>
                <h6 className='m-0'>Jobs</h6>
            </CardHeader>
            <CardBody className='p-0 pb-3 overflow-auto'>
                <table className='table mb-0'>
                    <thead className='bg-light'>
                        <tr>
                            <th scope='col' className='border-0'>Company Name</th>
                            <th scope='col' className='border-0'>Role</th>
                            <th scope='col' className='border-0'>Location</th>
                            <th scope='col' className='border-0'>Type</th>
                            <th scope='col' className='border-0'>Created At</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            jobs.map(job => (
                                <tr key={job.id} onClick={() => jobClicked(job.id)}>
                                    <td>{job.company}</td>
                                    <td>{job.title}</td>
                                    <td>{job.location}</td>
                                    <td>{job.type}</td>
                                    <td>{new Date(job.created_at).toDateString()}</td>
                                </tr>
                            ))
                        }
                    </tbody>
                </table>
            </CardBody>
        </Card>
    </Col>
  )
}

Table.propTypes = {
  jobs: PropTypes.array,
  history: PropTypes.func
}

export default withRouter(Table)
