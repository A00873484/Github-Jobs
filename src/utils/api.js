import appendToGet from './appendToGet'
import axios from 'axios'

export async function searchForJobs ({ description, fulltime, location }) {
  const res = await axios.get(`http://localhost:3001/github/jobs${appendToGet({ description, location, full_time: fulltime })}`)
  return res.data
}

export async function searchForJob (id) {
  const res = await axios.get(`http://localhost:3001/github/job?id=${id}`)
  return res.data
}
