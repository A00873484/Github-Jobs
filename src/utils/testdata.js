export const DataObject = {    
  "3c5fc02a-159b-4930-92aa-16bbc436f98a": {
    company: "Snickfish",
    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpb…ca60bf3947b4d117de93ba82e7b031fb2dc/snickfish-g-small.fw.png",
    company_url: "http://Snickfish",
    created_at: "Sat Feb 20 12:38:44 UTC 2021",
    description: "<p>LOOKING FOR LOCAL PHP DEVELOPER IN THE KNOXVILLE/MARYVILLE, TENNESSEE AREA.\nWe are a title management company based in Maryville, TN.  We service over 41 states across the U.S. and are looking for someone with an upbeat professional attitude who's eager to hang with us.  It's an exciting and fun place to work where you can learn, earn, and grow with us! This position is NOT a remote position. Please do NOT apply if you cannot work at our offices in Maryville, Tennessee.</p>\n<p>PHP DEVELOPER\nQualified, self-driven developer responsible for maintaining, updating, and upgrading client and server-side HTML /CSS /JS, PHP &amp; Node Systems.\nResponsibilities include setting deadlines, performing or automating regular maintenance tasks, performing system account updates, and optimizing existing systems in both PHP 5.6 and 7.2.\nPay is relevant to work experience and knowledge base.</p>\n<p>General Requirements\nExperienced in Large or Scalable Web Development\nSeasoned in PHP\nWork Well in a fast-paced environment\nMust be able to use high-level logic &amp; reasoning\nMust have excellent communication skill\nMust be able to plan and set deadlines effectively\nMust be willing to learn, and keep up to date knowledge on web development practices\nKnowledgeable in Standard HTML, CSS &amp; JS\nKnowledgeable in Objective, Functional, &amp; Procedural Programing &amp; Methodologies\nAWS Experience (including but not limited to EC2, Route 53,  EBS operational &amp; maintenance experience)\nMySQL General Experience\nDocker &amp; Containerization knowledge, and experience.\n5 + years general programing experience\n2 + years of Object-Oriented programing experience\nLanguage/ Framework /Library Requirements:\nPHP 5.6\nPHP 7.3\nNodeJS (JavaScript)\nMySQL\nComposer (PHP)\nNPM (NodeJS)\njQuery (JavaScript)\nBootstrap (HTML, CSS)\nRecommended Frameworks/Libraries\nLaravel (PHP)\nReact (NodeJS)\nEnvironment Familiarity (Required)\nDocker\nAWS EC2\nLinux</p>\n<p>We provide a nice friendly environment with professional people.\nWe hire for life.\nSome staffers have been with us since our conception 13 years ago. Our employees arrive to work happy and leave happy and we like that a lot.\nThe dress code is casual.\nYou will receive vacation pay, major holiday pay, PTO, and bonuses.</p>\n<p>Hours are 8-5 Monday-Friday, with evenings and weekends off.\nSend us your resume and tell us a little about yourself, and we'll contact you with further details about our company and how to schedule an interview.</p>\n",
    how_to_apply: "<p>Email your resume to <a href=\"mailto:NH8570@gmail.com\">NH8570@gmail.com</a></p>\n",
    id: "3c5fc02a-159b-4930-92aa-16bbc436f98a",
    location: "Maryville, Tennessee",
    title: "PHP Developer - NOT A REMOTE POSITION",
    type: "Full Time",
    url: "https://jobs.github.com/positions/3c5fc02a-159b-4930-92aa-16bbc436f98a"
  }, 
  "07d9afe8-0916-40f5-a508-2a42d9dc6c98": {
    company: "brightwheel ",
    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpb…lkIn19--8665ccd64baddd8e0766413744ed285dbe693bf1/Logo%20.png",
    company_url: "https://mybrightwheel.com/careers/",
    created_at: "Mon Feb 22 16:45:45 UTC 2021",
    description: "<p>Our Mission and Opportunity\nEarly education is one of the greatest determinants of childhood outcomes, is a must for working families, and has lasting social and economic impact. Brightwheel’s mission is to transform early education for every teacher, child, and parent — by giving teachers meaningfully more time with students each day, engaging parents in the development of their kids, and supporting the small businesses that make up the backbone of the $100 billion early education market. Brightwheel is the most loved technology brand in early education globally, trusted by thousands of educators and millions of families.</p>\n<p>Our Team\nWe are a distributed team, with offices in San Francisco, Denver, and Austin along with fully remote team members. Our team is passionate, talented, and customer-focused. Our exceptional investor group includes Addition, Bessemer Venture Partners, Chan Zuckerberg Initiative, GGV Capital, Lowercase Capital, Emerson Collective, and Mark Cuban.</p>\n<p>We believe that everyone—from our employees to the students, teachers, and administrators we serve— should be given the opportunity to learn and thrive, whatever their background may be. We celebrate diversity in all forms because it allows our team and the communities we serve to reach their full potential and do their best work.</p>\n<p>Who You Are\nWe are seeking a senior software engineer who is not only well-rounded technically, but also someone who is passionate about the positive impact their code can make in people's lives. Brightwheel engineers work closely with product and design, as well as with other areas of the business to deliver a platform that meets our teachers’ &amp; parents’ biggest needs. Every engineer at brightwheel is afforded a high degree of ownership and autonomy. We are a lean, fast-moving team and value individuals who can optimize for being impactful while considering both technical and business drivers.</p>\n<p>What You'll Do\n-Work alongside product and design to scope and implement new features or product lines.\n-Be a steward of quality, scalability, security, and performance. You'll work with other engineers to ensure that we have a solid foundation that serves our customers, and enables the team to continue building a great product.\n-Design and architect new software systems, or introduce new technologies, as appropriate to support brightwheel’s growth.\n-Share knowledge and expertise across the engineering team</p>\n<p>Qualifications, Skills, &amp; Abilities\n-4+ years of professional experience, preferably in dynamic companies/teams\n-Expertise in one or more application programming languages (e.g Ruby, JavaScript or Python)\n-Proficiency in building and debugging distributed systems on AWS\n-Proficiency in both contributing to and designing APIs</p>\n<p>Candidates with varying backgrounds are encouraged to apply. We do not require prior knowledge of technologies in our stack, only a desire to learn them. At brightwheel you will encounter:</p>\n<ul>\n<li>React.js</li>\n<li>Ruby on Rails</li>\n<li>PostgreSQL</li>\n<li>Redis</li>\n<li>CircleCI</li>\n<li>Kubernetes</li>\n<li>AWS managed services, like Redshift and ECS</li>\n</ul>\n<p>Brightwheel is proud to celebrate diversity and is committed to building an inclusive workplace regardless of race, color, religion, sex (including pregnancy and gender identity), national origin, political affiliation, sexual orientation, marital status, disability, genetic information, age, membership in an employee organization, retaliation, parental status, military service, or other non-merit factor.</p>\n",
    how_to_apply: "<p>Please apply using the link below:</p>\n<p><a href=\"https://jobs.lever.co/brightwheel/6e96774f-8350-495d-a567-c7…ba55f171d?lever-origin=applied&amp;lever-source%5B%5D=GitHub\">https://jobs.lever.co/brightwheel/6e96774f-8350-495d-a567-c7eba55f171d?lever-origin=applied&amp;lever-source%5B%5D=GitHub</a></p>\n",
    id: "07d9afe8-0916-40f5-a508-2a42d9dc6c98",
    location: "Anywhere in the US",
    title: "Senior Software Engineer, Backend",
    type: "Full Time",
    url: "https://jobs.github.com/positions/07d9afe8-0916-40f5-a508-2a42d9dc6c98",
  }
}
export const DataArray = [
  {
    company: "Snickfish",
    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpb…ca60bf3947b4d117de93ba82e7b031fb2dc/snickfish-g-small.fw.png",
    company_url: "http://Snickfish",
    created_at: "Sat Feb 20 12:38:44 UTC 2021",
    description: "<p>LOOKING FOR LOCAL PHP DEVELOPER IN THE KNOXVILLE/MARYVILLE, TENNESSEE AREA.\nWe are a title management company based in Maryville, TN.  We service over 41 states across the U.S. and are looking for someone with an upbeat professional attitude who's eager to hang with us.  It's an exciting and fun place to work where you can learn, earn, and grow with us! This position is NOT a remote position. Please do NOT apply if you cannot work at our offices in Maryville, Tennessee.</p>\n<p>PHP DEVELOPER\nQualified, self-driven developer responsible for maintaining, updating, and upgrading client and server-side HTML /CSS /JS, PHP &amp; Node Systems.\nResponsibilities include setting deadlines, performing or automating regular maintenance tasks, performing system account updates, and optimizing existing systems in both PHP 5.6 and 7.2.\nPay is relevant to work experience and knowledge base.</p>\n<p>General Requirements\nExperienced in Large or Scalable Web Development\nSeasoned in PHP\nWork Well in a fast-paced environment\nMust be able to use high-level logic &amp; reasoning\nMust have excellent communication skill\nMust be able to plan and set deadlines effectively\nMust be willing to learn, and keep up to date knowledge on web development practices\nKnowledgeable in Standard HTML, CSS &amp; JS\nKnowledgeable in Objective, Functional, &amp; Procedural Programing &amp; Methodologies\nAWS Experience (including but not limited to EC2, Route 53,  EBS operational &amp; maintenance experience)\nMySQL General Experience\nDocker &amp; Containerization knowledge, and experience.\n5 + years general programing experience\n2 + years of Object-Oriented programing experience\nLanguage/ Framework /Library Requirements:\nPHP 5.6\nPHP 7.3\nNodeJS (JavaScript)\nMySQL\nComposer (PHP)\nNPM (NodeJS)\njQuery (JavaScript)\nBootstrap (HTML, CSS)\nRecommended Frameworks/Libraries\nLaravel (PHP)\nReact (NodeJS)\nEnvironment Familiarity (Required)\nDocker\nAWS EC2\nLinux</p>\n<p>We provide a nice friendly environment with professional people.\nWe hire for life.\nSome staffers have been with us since our conception 13 years ago. Our employees arrive to work happy and leave happy and we like that a lot.\nThe dress code is casual.\nYou will receive vacation pay, major holiday pay, PTO, and bonuses.</p>\n<p>Hours are 8-5 Monday-Friday, with evenings and weekends off.\nSend us your resume and tell us a little about yourself, and we'll contact you with further details about our company and how to schedule an interview.</p>\n",
    how_to_apply: "<p>Email your resume to <a href=\"mailto:NH8570@gmail.com\">NH8570@gmail.com</a></p>\n",
    id: "3c5fc02a-159b-4930-92aa-16bbc436f98a",
    location: "Maryville, Tennessee",
    title: "PHP Developer - NOT A REMOTE POSITION",
    type: "Full Time",
    url: "https://jobs.github.com/positions/3c5fc02a-159b-4930-92aa-16bbc436f98a",
  },
  {
    company: "brightwheel ",
    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpb…lkIn19--8665ccd64baddd8e0766413744ed285dbe693bf1/Logo%20.png",
    company_url: "https://mybrightwheel.com/careers/",
    created_at: "Mon Feb 22 16:45:45 UTC 2021",
    description: "<p>Our Mission and Opportunity\nEarly education is one of the greatest determinants of childhood outcomes, is a must for working families, and has lasting social and economic impact. Brightwheel’s mission is to transform early education for every teacher, child, and parent — by giving teachers meaningfully more time with students each day, engaging parents in the development of their kids, and supporting the small businesses that make up the backbone of the $100 billion early education market. Brightwheel is the most loved technology brand in early education globally, trusted by thousands of educators and millions of families.</p>\n<p>Our Team\nWe are a distributed team, with offices in San Francisco, Denver, and Austin along with fully remote team members. Our team is passionate, talented, and customer-focused. Our exceptional investor group includes Addition, Bessemer Venture Partners, Chan Zuckerberg Initiative, GGV Capital, Lowercase Capital, Emerson Collective, and Mark Cuban.</p>\n<p>We believe that everyone—from our employees to the students, teachers, and administrators we serve— should be given the opportunity to learn and thrive, whatever their background may be. We celebrate diversity in all forms because it allows our team and the communities we serve to reach their full potential and do their best work.</p>\n<p>Who You Are\nWe are seeking a senior software engineer who is not only well-rounded technically, but also someone who is passionate about the positive impact their code can make in people's lives. Brightwheel engineers work closely with product and design, as well as with other areas of the business to deliver a platform that meets our teachers’ &amp; parents’ biggest needs. Every engineer at brightwheel is afforded a high degree of ownership and autonomy. We are a lean, fast-moving team and value individuals who can optimize for being impactful while considering both technical and business drivers.</p>\n<p>What You'll Do\n-Work alongside product and design to scope and implement new features or product lines.\n-Be a steward of quality, scalability, security, and performance. You'll work with other engineers to ensure that we have a solid foundation that serves our customers, and enables the team to continue building a great product.\n-Design and architect new software systems, or introduce new technologies, as appropriate to support brightwheel’s growth.\n-Share knowledge and expertise across the engineering team</p>\n<p>Qualifications, Skills, &amp; Abilities\n-4+ years of professional experience, preferably in dynamic companies/teams\n-Expertise in one or more application programming languages (e.g Ruby, JavaScript or Python)\n-Proficiency in building and debugging distributed systems on AWS\n-Proficiency in both contributing to and designing APIs</p>\n<p>Candidates with varying backgrounds are encouraged to apply. We do not require prior knowledge of technologies in our stack, only a desire to learn them. At brightwheel you will encounter:</p>\n<ul>\n<li>React.js</li>\n<li>Ruby on Rails</li>\n<li>PostgreSQL</li>\n<li>Redis</li>\n<li>CircleCI</li>\n<li>Kubernetes</li>\n<li>AWS managed services, like Redshift and ECS</li>\n</ul>\n<p>Brightwheel is proud to celebrate diversity and is committed to building an inclusive workplace regardless of race, color, religion, sex (including pregnancy and gender identity), national origin, political affiliation, sexual orientation, marital status, disability, genetic information, age, membership in an employee organization, retaliation, parental status, military service, or other non-merit factor.</p>\n",
    how_to_apply: "<p>Please apply using the link below:</p>\n<p><a href=\"https://jobs.lever.co/brightwheel/6e96774f-8350-495d-a567-c7…ba55f171d?lever-origin=applied&amp;lever-source%5B%5D=GitHub\">https://jobs.lever.co/brightwheel/6e96774f-8350-495d-a567-c7eba55f171d?lever-origin=applied&amp;lever-source%5B%5D=GitHub</a></p>\n",
    id: "07d9afe8-0916-40f5-a508-2a42d9dc6c98",
    location: "Anywhere in the US",
    title: "Senior Software Engineer, Backend",
    type: "Full Time",
    url: "https://jobs.github.com/positions/07d9afe8-0916-40f5-a508-2a42d9dc6c98",
  }
]
export const DataInstance = DataObject["3c5fc02a-159b-4930-92aa-16bbc436f98a"]
