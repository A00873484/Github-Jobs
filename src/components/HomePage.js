import React from 'react'
import { Container, Row } from 'shards-react'
import JobsTable from './JobsTable'
import InputBox from './InputBox'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import PageTitle from './common/PageTitle'

function HomePage ({ jobs }) {
  return (
    <div>
      <Container fluid className='main-content-container px-4'>
      <Row noGutters className='page-header py-4'>
        <PageTitle title='Search' subtitle='Github Jobs' className='text-sm-left mb-3' />
      </Row>
        <Row>
          <InputBox/>
        </Row>
        <Row noGutters className='page-header py-4'>
          {
            jobs.length > 0 && <JobsTable jobs={jobs} />
          }
        </Row>
      </Container>
    </div>
  )
}

HomePage.propTypes = {
  jobs: PropTypes.array
}

function mapStateToProps ({ jobs = [] }) {
  console.log(jobs)
  return {
    jobs
  }
}

export default connect(mapStateToProps)(HomePage)
