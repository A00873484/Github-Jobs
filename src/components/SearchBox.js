import React, { useState } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import {
  Card,
  CardBody,
  Form,
  Row,
  Col,
  FormInput,
  Button
} from 'shards-react'
import { receiveJobs } from '../actions/jobs'
import { searchForJobs } from '../utils/api'
import { showLoading, hideLoading } from 'react-redux-loading-bar'

function InputBox ({ dispatch }) {
  const [search, setSearch] = useState('')
  const [location, setLocation] = useState('')
  const [fulltime, setFullTime] = useState(false)

  const searchClicked = (e) => {
    e.preventDefault()
    dispatch(showLoading())
    searchForJobs({ description: search, location, fulltime }).then(res => {
      dispatch(receiveJobs(res))
      dispatch(hideLoading())
    })
  }

  return <Card small className='mb-4 w-100'>
        <CardBody className='d-flex flex-column'>
            <Form onSubmit={searchClicked}>
                <Row form>
                    <Col md="8" className="form-group flex-grow-1">
                        <FormInput id='search' placeholder='Search' value={search} onChange={e => setSearch(e.target.value)}/>
                    </Col>
                    <Col md="4" className="form-group">
                        <FormInput id='location' placeholder='Location' value={location} onChange={e => setLocation(e.target.value)}/>
                    </Col>
                    <Col className='d-flex flex-row justify-content-between'>
                        <label className='align-self-center'>
                            <input id='fullTimeInputId' type='checkbox' onClick={() => setFullTime(!fulltime)}/>
                            <span className='pl-2'>Full Time Only</span>
                        </label>
                        <Button id='button-go' theme='accent' type='submit' className='align-self-center'>
                            Go
                        </Button>
                    </Col>
                </Row>
            </Form>
        </CardBody>
    </Card>
}

InputBox.propTypes = {
  dispatch: PropTypes.func
}

export default connect()(InputBox)
