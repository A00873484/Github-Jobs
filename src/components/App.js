import React, { Fragment } from 'react'
import HomePage from './HomePage'
import DetailsPage from './DetailsPage'
import { LoadingBar } from 'react-redux-loading'
import { BrowserRouter as Router, Route } from 'react-router-dom'

function App () {
  return (
    <Router>
      <Fragment>
        <LoadingBar/>
        <div className='container'>
          <Route path='/' exact component={HomePage}/>
          <Route path='/details/:id' exact component={DetailsPage}/>
        </div>
      </Fragment>
    </Router>
  )
}

export default App
