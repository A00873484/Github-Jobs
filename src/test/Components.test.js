import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../components/App'
import HomePage from '../components/HomePage'
import DetailsPage from '../components/DetailsPage'
import SearchBox from '../components/SearchBox'
import JobsTable from '../components/JobsTable'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { createMemoryHistory } from 'history'
import configureStore from 'redux-mock-store'
import { DataObject, DataArray, DataInstance } from '../utils/testdata'

const mockStore = configureStore()
const store = mockStore({ jobs: DataObject })
Enzyme.configure({ adapter: new Adapter() })

describe('SearchBox', () => {
  it('should contain 2 inputboxs, 1 checkbox, and 1 button ', () => {
    const wrapper = shallow(<SearchBox store={store} />);
    const component = wrapper.dive().dive()
    expect(component.find('#search').prop('placeholder')).toBe('Search')
    expect(component.find('#location').prop('placeholder')).toBe('Location')
    expect(component.find('span.pl-2').text()).toBe('Full Time Only')
    expect(component).toMatchSnapshot()
  })
})

describe('JobsTable', () => {
  const props = {}
  props.jobs = DataArray
  props.history = createMemoryHistory('/')
  const wrapper = shallow(<Router><JobsTable {...props}/></Router>)
  const component = wrapper.dive().dive().dive().dive().dive().dive()

  test('should contain passed in data', () => {
    // Has Title
    expect(component.find('h6.m-0').text()).toBe('Jobs')
    // First Set
    expect(component.find('tbody tr td').at(0).text()).toBe(DataArray[0].company)
    expect(component.find('tbody tr td').at(1).text()).toBe(DataArray[0].title)
    expect(component.find('tbody tr td').at(2).text()).toBe(DataArray[0].location)
    expect(component.find('tbody tr td').at(3).text()).toBe(DataArray[0].type)
    expect(component.find('tbody tr td').at(4).text()).toBe(new Date(DataArray[0].created_at).toDateString())
    // Second Set
    expect(component.find('tbody tr td').at(5).text()).toBe(DataArray[1].company)
    expect(component.find('tbody tr td').at(6).text()).toBe(DataArray[1].title)
    expect(component.find('tbody tr td').at(7).text()).toBe(DataArray[1].location)
    expect(component.find('tbody tr td').at(8).text()).toBe(DataArray[1].type)
    expect(component.find('tbody tr td').at(9).text()).toBe(new Date(DataArray[1].created_at).toDateString())
    expect(component).toMatchSnapshot()
  })
})

describe('DetailsPage', () => {
  const props = {match: {params: {id: "3c5fc02a-159b-4930-92aa-16bbc436f98a"}}}
  props.history = createMemoryHistory('/') 
  
  const wrapper = shallow(<DetailsPage {...props} store={store}/>)
  const component = wrapper.dive().dive()

  test('should contain passed in data', () => {
    expect(component.find('h3.mb-3').text()).toBe(DataInstance.company);
    expect(component.find('a.text-secondary').text()).toBe(DataInstance.company_url);
    expect(component.find('div#location').text()).toBe(DataInstance.location);
    expect(component.find('div#title').text()).toBe(DataInstance.title);
    expect(component.find('div#type').text()).toBe(DataInstance.type);
    expect(component.find('span#date').text()).toBe(new Date(DataInstance.created_at).toDateString());
    expect(component.find('img').prop('src')).toEqual(DataInstance.company_logo);
  })
})

