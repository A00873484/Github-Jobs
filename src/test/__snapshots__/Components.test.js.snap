// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsTable should contain passed in data 1`] = `ShallowWrapper {}`;

exports[`SearchBox should contain 2 inputboxs, 1 checkbox, and 1 button  1`] = `ShallowWrapper {}`;
